utils/anaoptions.py

```diff
@@ -11,15 +11,25 @@
 """
 
 import argparse
-parent_parser = argparse.ArgumentParser(add_help = False)
+parentParser1 = argparse.ArgumentParser(add_help = False)
 
 # Positional arguments
-parent_parser.add_argument("infilename",type=str,help="Specify Input Filename by full path")
+parentParser1.add_argument("infilename",type=str,help="Specify Input Filename by full path")
 
 # Optional arguments
-parent_parser.add_argument("-d", "--debug", action="store_true", help="print extra debugging information")
-parent_parser.add_argument("-o", "--outfilename", type=str, help="Specify Output Filename")
+parentParser1.add_argument("-d", "--debug", action="store_true", help="print extra debugging information")
+parentParser1.add_argument("-o", "--outfilename", type=str, help="Specify Output Filename")
 
-stripChanOrPinType = parent_parser.add_mutually_exclusive_group(required=False)
+stripChanOrPinType = parentParser1.add_mutually_exclusive_group(required=False)
 stripChanOrPinType.add_argument("-c","--channels", action="store_true", help="Make plots vs channels instead of strips")
 stripChanOrPinType.add_argument("-p","--panasonic", action="store_true", dest="PanPin",help="Make plots vs Panasonic pins instead of strips")
+
+parentParser2 = argparse.ArgumentParser(add_help = False)
```

Comment:
Previously the name `parent_parser` was a descriptive and unique variable name that all things "inherited" from; but by inserting a second one and changing to names `parentParser1` and `parentParser2` you have really degraded the overall readability of the code base; what I was going for more was a common sense copy/paste from `ana_scans.py`: e.g.

```python
    # create the parent parser for channel masks for scurve analysis	
    parser_scurveChanMasks = argparse.ArgumentParser(add_help = False)	

     chanMaskGroup = parser_scurveChanMasks.add_argument_group(title="Options for channel mask decisions", description="Parameters which specify how Dead, Noisy, and High Pedestal Channels are charaterized")	
    chanMaskGroup.add_argument("--maxEffPedPercent", type=float, default=0.02, help="Percentage, threshold for setting the HighEffPed mask reason, if channel (effPed > maxEffPedPercent * nevts) then HighEffPed is set")	
    chanMaskGroup.add_argument("--highNoiseCut", type=float, default=1.5, help="Threshold in fC for setting the HighNoise maskReason, if channel (scurve_sigma > highNoiseCut) then HighNoise is set")	
    chanMaskGroup.add_argument("--deadChanCutLow", type=float, default=0,help="If channel (deadChanCutLow < scurve_sigma < deadChanCutHigh) then DeadChannel is set")	
    chanMaskGroup.add_argument("--deadChanCutHigh", type=float, default=0, help="If channel (deadChanCutHigh < scurve_sigma < deadChanCutHigh) then DeadChannel is set")
```   

And then in the three files discussed to import `parser_scurveChanMasks` and add this to the list of parent parsers that are provided to various analysis tools.

Unfortunately you've rolled all your changes into a *single* commit which isn't compartmentalized so you cannot easily roll back these changes.

Please do the above and revert changes, e.g. `parentParser1` -> `parent_parser` then revert changes to files not mentioned in issue #235.  Then squash this change commit so that the history of those files you will revert is not affected; this might not be the correct `git` protocol; @jsturdy can better advise here.

Reply:
done